### PR TITLE
Add command line flags to teardown command 

### LINKED ISSUE #17: Add confirmation for tearing down a workflow. (Include a `-f` flag for auto-confirm)
Don't do #14 until this one has been completed. We need to differentiate between flag arguments and regular ones, as well as deal with varying amounts of arguments.

Whoever does this one should probably also do #14

### PR DESCRIPTION
**Pull request by johnisom**
With the changes that this PR introduces, one can provide the `-f` or `--force` boolean flags to the teardown command to bypass the confirmation prompt. One can also provide a `--roles` flag that will specify which IAM roles to also delete, in the form of
`--roles=<role1>,<role2>`. Right now, the `--roles` flag is still a WIP and shouldn't be considered ready for review

This PR fixes #17 and #14.

### CODE REVIEW

**Comment by johnisom** [comment]
@T-monius @zklamm if you two could give this a somewhat thorough review, that would be appreciated. There were a lot of changes made in this PR, some somewhat fundamental

File: bin/teardown.js [hunk 1]
```diff
@@ -2,33 +2,50 @@
...
-const lambdaNames = fs.readdirSync("lambdas").map(basename);
+const argv = minimist(process.argv.slice(2), {
+  boolean: ["f", "force"],
+  string: ["roles"],
+  default: {
+    roles: "",
+  },
+});
+const stateMachineName = argv._[0];
+const rolesToDelete = argv.roles.split(",").filter((role) => role.length > 0);
```
Comment (johnisom) [inline bin/teardown.js:25]: This is for command line argument parsing

File: bin/teardown.js [hunk 2]
```diff
@@ -2,33 +2,50 @@
...
+  const deleteLambdasPromise = deleteLambdas(lambdaNames).catch(console.log);
+  const deleteStateMachinePromise = getStateMachineArn(stateMachineName)
+    .then(deleteStateMachine)
+    .catch(console.log);
+  await Promise.all([deleteLambdasPromise, deleteStateMachinePromise]);
+
+  rolesToDelete.forEach(teardownRoleByName);
+};
+
+if (argv.force || argv.f) {
```
Comment (johnisom) [inline bin/teardown.js:44]: Thid is also command line argument parsing, checking if a `--force` or `-f` flag was provided

File: bin/teardown.js (hunk 1 above)
Comment (T-monius) [reply bin/teardown.js:25]: Nice

File: bin/teardown.js (hunk 2 above)
Comment (T-monius) [reply bin/teardown.js:44]: Good, this is just what we were looking to do.

**Comment by johnisom** [comment]
OK, I merged master into here, it looks good. I'll go ahead and merge it

**Pull request merged**